lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -723,17 +723,11 @@ static bool getFieldName(const FieldRef &fieldRef, SmallString<32> &string) {
 //===----------------------------------------------------------------------===//
 
 /// Check whether a type contains a `ResetType`.
-static bool typeContainsReset(FIRRTLType type) {
-  return TypeSwitch<FIRRTLType, bool>(type)
-      .Case<BundleType>([](auto type) {
-        for (auto e : type.getElements())
-          if (typeContainsReset(e.type))
-            return true;
-        return false;
+static bool typeContainsReset(Type type) {
```

Comment:
Feel free to peel this off and commit it directly.

Reply:
Nice!